protocols/secio/src/handshake.rs

```diff
@@ -490,15 +487,15 @@ where
                 #[cfg(feature = "secp256k1")]
                 PublicKey::Secp256k1(ref remote_public_key) => {
                     let data_to_verify = Sha256::digest(&data_to_verify);
-                    let message = secp256k1::Message::from_slice(data_to_verify.as_ref())
-                        .expect("digest output length doesn't match secp256k1 input length");
-                    let secp256k1 = secp256k1::Secp256k1::verification_only();
-                    let signature = secp256k1::Signature::from_der(&secp256k1, remote_exch.get_signature());
-                    let remote_public_key = secp256k1::key::PublicKey::from_slice(&secp256k1, remote_public_key);
+                    let message = secp256k1::Message::parse_slice(&data_to_verify)?;
+                    let signature = secp256k1::Signature::parse_der(remote_exch.get_signature());
+                    // TODO: is the key compressed or not?
+                    let remote_public_key = secp256k1::PublicKey::parse_slice(remote_public_key, None);
+
                     if let (Ok(signature), Ok(remote_public_key)) = (signature, remote_public_key) {
-                        match secp256k1.verify(&message, &signature, &remote_public_key) {
-                            Ok(()) => (),
-                            Err(_) => {
+                        match secp256k1::verify(&message, &signature, &remote_public_key) {
+                            true => (),
+                            false => {
```

Comment:
Why not `if !secp256k1::verify(&message, &signature, &remote_public_key) { ... }`?